workspaces/optic-engine/src/shapes/traverser.rs

```diff
@@ -287,6 +287,10 @@ impl ShapeTrail {
     new_trail.path.push(component);
     new_trail
   }
+
+  pub fn last_component(&self) -> Option<&ShapeTrailPathComponent> {
+    self.path.last()
```

Comment:
do you ever want the root to come out here?

Reply:
Yeah, why not? 🤷‍♂️ It seems to me this would return whatever the last component of the shape trail is, and if that's the root, sure? I don' think there's enough context from this method to limit that, that's up to the caller to decide.